### ISSUE TITLE
Replace two build configurations with dynamic method replacement

### ISSUE THREAD

**Issue by Scooletz** [Scooletz/RampUp#6]
Currently `AtomicLong` and `AtomicInt` are built conditionally, with two different methods` implementations, one targeted at mocking for tests, the other, for real use. If we could replace the body on the flight, when testing, that would result in having a single `Release` configuration.
You could consider using an approach from [ReJit](https://github.com/damageboy/ReJit) obtaining the method body ptr and overwriting it, but consider the fact that the mocked implementations have longer bodies and you can't get the compiled method body lenght I suppose.
You need to get dirty with method descriptors, pointers, JIT but it should be possible after some research.

[CodeCop](http://getcodecop.com/) looks like the way to do it. Even with limitation of 25 methods replaced. 
[EasyHook](https://easyhook.github.io/) could be considered when first isn't suitable.

**Comment by Scooletz** [comment]
Waiting for a fixup of the bug isolated in here: https://gist.github.com/Scooletz/de7f6b63bce5f51b9053 and discussed in here: https://twitter.com/Scooletz/status/704773478705188866
Once it's done, this will be an easy thing to fix.

**Comment by danielmarbach** [comment]
How about https://github.com/Fody/Ionad?

**Comment by Scooletz** [comment]
@danielmarbach It won't work. This covers `static` and requires build. My aim is to have one `Release` mode and replace methods dynamically only when run in tests. Sth similar is done by [ReJit](https://github.com/damageboy/ReJit), with preJITting methods and then replacing method bodies with a compiled assembler. In my case replacements are bigger than original methods, hence, without messing with method descriptors and putting unconditional jump in ASM I see no way of doing it. 
Another case would be using Cecil and rewrite assembly and load it from `rawBytes`, but this approach didn't work (I've tried).
CodeCop seems to be solving that. I know that they're working on fixing the bug I've found, so I'll give them some time ;-)
